server/src/main/java/org/elasticsearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -152,6 +152,7 @@ private void buildResponse(final ClusterStateRequest request,
         MetaData.Builder mdBuilder = MetaData.builder();
         mdBuilder.clusterUUID(currentState.metaData().clusterUUID());
         mdBuilder.coordinationMetaData(currentState.coordinationMetaData());
+        mdBuilder.version(currentState.metaData().version());
```

Comment:
why always return it, even if the metadata was not requested?

Reply:
I feel that the "version" belongs the metadata group of the metadata just like "clusterUUID".

Reply:
The cluster uuid is exceptional (because it used in many places for identification purposes), the metadata version not so much (imo), as there is also the cluster state version (that is persisted starting in 7.0). I think we should move `mdBuilder.version(currentState.metaData().version());` below `if (request.indices().length > 0) {`

Reply:
I am ok with your suggestion :).